lib/src/embedded/compiler.ts

```diff
@@ -0,0 +1,79 @@
+// Copyright 2020 Google Inc. Use of this source code is governed by an
+// MIT-style license that can be found in the LICENSE file or at
+// https://opensource.org/licenses/MIT.
+
+import {InboundMessage, OutboundMessage} from '../vendor/embedded_sass_pb';
+import {EmbeddedProcess} from './compiler/process';
+import {PacketTransformer} from './compiler/packet-transformer';
+import {MessageTransformer} from './compiler/message-transformer';
+import {Observable, Subject, merge} from 'rxjs';
+import {filter, first, map} from 'rxjs/operators';
+
+/**
+ * A facade to the Embedded Sass Compiler.
+ *
+ * Takes the compiler's raw stdin and stdout and transforms them into public
+ * Observables of InboundMessages (write$) and OutboundMessages (read$). Also
+ * exposes an Observable (error$) that emits if a ProtocolError occurs while
+ * communicating with the compiler.
+ *
+ * If there is a ProtocolError, this shuts down the compiler process and cleans
+ * up all Observables.
+ */
+export class Compiler {
```

Comment:
I suspect at some point we're going to want to expose a public class called `Compiler` to address https://github.com/sass/sass/issues/2745, so it might be worth finding a different name for this now so we don't have to rename it later. Maybe `Host`?

Reply:
Good point, thanks for calling this out